main.tf

```
@@ -50,20 +54,49 @@ locals {
     "ecr:UploadLayerPart",
   ]
 
-  push_statement = length(var.push_identities) > 0 ? [{
+  # ---------------------------------------------------------------------------------------------------------------------
+  # (CROSS ACCOUNT IDENTITIES ACCESS
+  # Grant identities such as IAM users pull and/or push access
+  # to the repositories.
+  # ---------------------------------------------------------------------------------------------------------------------
+  pull_identites_statement = length(var.pull_identities) > 0 ? [{
+    actions     = local.ecr_pull_actions
+    identifiers = var.pull_identities
+  }] : []
+
+  push_identites_statement = length(var.push_identities) > 0 ? [{
     actions     = local.ecr_push_actions
     identifiers = var.push_identities
   }] : []
 
-  pull_statement = length(var.pull_identities) > 0 ? [{
-    actions     = local.ecr_pull_actions
-    identifiers = var.pull_identities
+  # ---------------------------------------------------------------------------------------------------------------------
+  # CROSS ACCOUNT ACCESS
+  # Account ids provided in 'pull_accounts' and 'push_accounts' will be added
+  # as principals with their related actions.
+  # Furthermore, the "ecr:GetAuthorizationToken" action to allow cross account
+  # authentication with ECR will be added.
+  # ---------------------------------------------------------------------------------------------------------------------
+  pull_cross_account_statement = length(var.pull_accounts) > 0 ? [{
+    actions = concat(
+      local.ecr_pull_actions,
+      local.ecr_authentication_actions
+    )
+    identifiers = formatlist("arn:aws:iam::%s:root", var.pull_accounts)
```

same as 

```suggestion
    identifiers = var.pull_accounts
```

afaik

---

nope?

---

ah cool wasn't aware of that :)

---

resolved in aebaa55